homeassistant/components/airly/__init__.py

```diff
@@ -49,43 +53,75 @@
 
     websession = async_get_clientsession(hass)
 
-    airly = AirlyData(websession, api_key, latitude, longitude)
+    # We check how many Airly config entries are and calculate interval to not
+    # exceed allowed numbers of requests.
+    update_interval = timedelta(
+        minutes=ceil(24 * 60 / MAX_REQUESTS_PER_DAY) * instances(hass)
+    )
+
+    coordinator = AirlyDataUpdateCoordinator(
+        hass, websession, api_key, latitude, longitude, update_interval
+    )
+    await coordinator.async_refresh()
 
-    await airly.async_update()
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][config_entry.entry_id] = coordinator
 
-    hass.data[DOMAIN][DATA_CLIENT][config_entry.entry_id] = airly
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, component)
+        )
+
+    # Change update_interval for another Airly instances
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.entry_id != config_entry.entry_id and hass.data.get(DOMAIN).get(
+            entry.entry_id
+        ):
+            hass.data.get(DOMAIN).get(entry.entry_id).update_interval = update_interval
```

Comment:
You know the keys exist so use square brackets.
```suggestion
            hass.data[DOMAIN][entry.entry_id].update_interval = update_interval
```